### PR TITLE
Update plugin to 4.0.0 to work for tslint@4.0.0

### PR DESCRIPTION
**Pull request by Izzmo**

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/grunt-tslint](https://github.com/palantir/grunt-tslint), @Izzmo! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by Izzmo** [comment]
@palantirtech done! Thanks.

**Comment by adidahiya** [comment]
@nchen63 can you please take a look

File: tasks/tslint.js [hunk 1]
```diff
@@ -50,9 +51,14 @@ module.exports = function (grunt) {
...
 
-                var contents = grunt.file.read(filepath);
-                var linter = new Linter(filepath, contents, options);
-                var result = linter.lint();
+                var lintOptions = {
+                    fix: options.fix,
+                    formatter: options.formatter
+                }
+
+                var linter = new Linter.Linter(lintOptions, Linter.Linter.createProgram(configuration.path));
```
Comment (adidahiya) [inline tasks/tslint.js]: this doesn't look quite right; this ends up creating a new program for every file in the input glob. a TypeScript "program" should only be created when that option is specified in the TSLint API.

**Comment by helfer** [comment]
@Izzmo are you still working on this PR?

**Comment by Izzmo** [comment]
@helfer yes sorry I'll finish it up tonight! The holidays got ahead of me.

**Comment by bondz** [comment]
@izzmo any updates?

**Comment by Izzmo** [comment]
Sorry guys, I'm having a heck of a time getting the tests to work. It would seem these haven't worked in a while though... so what should we do? @helfer @bondz

**Comment by Izzmo** [comment]
@helfer @adidahiya I updated the file to not create a program. Again, half the tests work.. half don't. I think this is basically because no one has kept them up to date with new dependency versions and node itself.

**Comment by bondz** [comment]
Ping @nchen63

**Pull request merged**